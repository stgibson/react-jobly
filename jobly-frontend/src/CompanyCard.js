import React from "react";
import Card from "react-bootstrap/Card";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

/**
 * Component for showing info on a company in a card
 * @param {Object{Object{string|number|Object{number|string}}}} param0
 * @returns JSX code for rendering company card
 */
const CompanyCard = ({ company }) => {
  // Learned how to use local image at https://www.freecodecamp.org/news/react-background-image-tutorial-how-to-set-backgroundimage-with-inline-css-style/
  return (
    <Card data-testid={ company.handle }>
      <Card.Body>
        <Card.Title>
          <Row>
            <Col xs={ 10 }>
              { company.name }
            </Col>
            {
              company.logoUrl &&
              <Col xs={ 2 }>
                <div
                  style={
                    { width: "50px", content: `url(${company.logoUrl})` }
                  }
                  data-testid={ company.logoUrl }
                >
                  &nbsp;
                </div>
              </Col>
            }
          </Row>
        </Card.Title>
        <Card.Text>{ company.description }</Card.Text>
      </Card.Body>
    </Card>
  );
};

export default CompanyCard;