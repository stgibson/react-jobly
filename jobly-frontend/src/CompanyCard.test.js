import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";

const company = {
  handle: "test",
  name: "Test",
  description: "Test Description",
  numEmployees: 100,
  logoUrl: "/logos/logo1.png"
};

const findAllCompanies = jest.fn(() => null); // Learned how to mock functions at https://jestjs.io/docs/mock-functions

it("renders without crashing", () => {
  render(<MemoryRouter><CompanyCard company={ company } /></MemoryRouter>);
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter><CompanyCard company={ company } /></MemoryRouter>
  );
  
  expect(asFragment()).toMatchSnapshot();
});

it("shows company details", () => {
  const { queryByText, queryByTestId } = render(
    <MemoryRouter><CompanyCard company={ company } /></MemoryRouter>
  );

  expect(queryByText(company.name)).toBeInTheDocument();
  expect(queryByText(company.description)).toBeInTheDocument();
  expect(queryByTestId(company.logoUrl)).toBeInTheDocument();
});