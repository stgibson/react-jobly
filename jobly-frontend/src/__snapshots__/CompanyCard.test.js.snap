// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="test"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        <div
          class="row"
        >
          <div
            class="col-10"
          >
            Test
          </div>
          <div
            class="col-2"
          >
            <div
              data-testid="/logos/logo1.png"
              style="width: 50px; content: url(/logos/logo1.png);"
            >
              Â 
            </div>
          </div>
        </div>
      </div>
      <p
        class="card-text"
      >
        Test Description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
